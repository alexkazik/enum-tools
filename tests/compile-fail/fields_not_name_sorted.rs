#![no_implicit_prelude]

use ::enum_tools::EnumTools;

#[derive(EnumTools, Debug, Clone, Copy, Eq, PartialEq)]
#[enum_tools(sorted(name))]
#[repr(i8)]
pub(crate) enum EG {
    #[enum_tools(rename = "Z")]
    A,
    B,
    C,
    D,
}

fn main() {}
