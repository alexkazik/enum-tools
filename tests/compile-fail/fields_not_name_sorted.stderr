error: variants are not sorted by name
  --> tests/compile-fail/fields_not_name_sorted.rs:11:5
   |
11 |     B,
   |     ^
